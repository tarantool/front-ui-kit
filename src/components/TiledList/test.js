import React from 'react';
import renderer from 'react-test-renderer';
import { css } from '@emotion/css';

import { Text, TiledList, TiledListItem } from '../../index';

describe('TiledList', () => {
  it('renders correctly', () => {
    const styles = {
      row: css`
        display: flex;
        justify-content: space-between;
      `,
      wrap: css`
        padding: 40px;
      `,
    };

    const items = [
      { name: 'John', balance: 200 },
      { name: 'Emily', balance: 1500 },
      { name: 'Michael', balance: 500 },
    ];

    const tree = renderer
      .create(
        <>
          <TiledList>
            {items.map((item) => (
              <TiledListItem className={styles.row} key={item.name}>
                <Text>{item.name}</Text>
                <Text variant="h5" tag="span">{`$${item.balance}`}</Text>
              </TiledListItem>
            ))}
          </TiledList>
          <TiledList outer={false}>
            {items.map((item) => (
              <TiledListItem className={styles.row} key={item.name} corners="soft">
                <Text>{item.name}</Text>
                <Text variant="h5" tag="span">{`$${item.balance}`}</Text>
              </TiledListItem>
            ))}
          </TiledList>
        </>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
