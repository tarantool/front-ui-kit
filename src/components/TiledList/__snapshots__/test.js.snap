// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TiledList renders correctly 1`] = `
Array [
  .emotion-0 {
  padding: 0;
  list-style: none;
  margin: 0 -16px;
}

.emotion-1 {
  padding: 12px 16px;
  margin-bottom: 8px;
  border-radius: 2px;
  background-color: #ffffff;
  box-shadow: 0 1px 4px 0 rgba(0, 0, 0, 0.11);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
}

.emotion-1:last-child {
  margin-bottom: 0;
}

.emotion-2 {
  margin: 0;
  font-family: Inter,Arial,sans-serif;
  font-size: 14px;
  line-height: 22px;
  color: #000000;
  -webkit-font-smoothing: antialiased;
}

.emotion-2 b {
  font-weight: 600;
}

.emotion-3 {
  margin: 0;
  font-family: Inter,Arial,sans-serif;
  font-size: 12px;
  line-height: 22px;
  text-transform: uppercase;
  letter-spacing: 0.01em;
  color: #040b1d;
  font-weight: 600;
  -webkit-font-smoothing: antialiased;
}

<ul
    className="emotion-0"
  >
    <li
      className="emotion-1"
    >
      <span
        className="emotion-2"
      >
        John
      </span>
      <span
        className="emotion-3"
      >
        $200
      </span>
    </li>
    <li
      className="emotion-1"
    >
      <span
        className="emotion-2"
      >
        Emily
      </span>
      <span
        className="emotion-3"
      >
        $1500
      </span>
    </li>
    <li
      className="emotion-1"
    >
      <span
        className="emotion-2"
      >
        Michael
      </span>
      <span
        className="emotion-3"
      >
        $500
      </span>
    </li>
  </ul>,
  .emotion-0 {
  padding: 0;
  list-style: none;
}

.emotion-1 {
  padding: 12px 16px;
  margin-bottom: 8px;
  border-radius: 2px;
  background-color: #ffffff;
  box-shadow: 0 1px 4px 0 rgba(0, 0, 0, 0.11);
  border-radius: 4px;
  margin-bottom: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
}

.emotion-1:last-child {
  margin-bottom: 0;
}

.emotion-2 {
  margin: 0;
  font-family: Inter,Arial,sans-serif;
  font-size: 14px;
  line-height: 22px;
  color: #000000;
  -webkit-font-smoothing: antialiased;
}

.emotion-2 b {
  font-weight: 600;
}

.emotion-3 {
  margin: 0;
  font-family: Inter,Arial,sans-serif;
  font-size: 12px;
  line-height: 22px;
  text-transform: uppercase;
  letter-spacing: 0.01em;
  color: #040b1d;
  font-weight: 600;
  -webkit-font-smoothing: antialiased;
}

<ul
    className="emotion-0"
  >
    <li
      className="emotion-1"
    >
      <span
        className="emotion-2"
      >
        John
      </span>
      <span
        className="emotion-3"
      >
        $200
      </span>
    </li>
    <li
      className="emotion-1"
    >
      <span
        className="emotion-2"
      >
        Emily
      </span>
      <span
        className="emotion-3"
      >
        $1500
      </span>
    </li>
    <li
      className="emotion-1"
    >
      <span
        className="emotion-2"
      >
        Michael
      </span>
      <span
        className="emotion-3"
      >
        $500
      </span>
    </li>
  </ul>,
]
`;
